src/averageAges.js

```diff
@@ -55,7 +68,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenArray = people.filter(child => onlyWithSon
+    ? (people.find(mother => child.mother === mother.name
+      && child.sex === 'm'))
+    : (people.find(mother => child.mother === mother.name)));
```

Comment:
```suggestion
  const children = people.filter(child => onlyWithSon
    ? (people.find(mother => child.mother === mother.name
      && child.sex === 'm'))
    : (people.find(mother => child.mother === mother.name)));
```

Reply:
Don't use names of data types or data structures in the names of your variables.